service/history/replication/task_ack_manager.go

```diff
@@ -155,14 +156,23 @@ func (t *taskAckManagerImpl) GetTasks(
 	readLevel := lastReadTaskID
 TaskInfoLoop:
 	for _, taskInfo := range taskInfoList {
+		// filter task info by domain clusters.
+		domainEntity, err := t.shard.GetDomainCache().GetDomainByID(taskInfo.GetDomainID())
```

Comment:
do we assume here the domain from the db is always in the cache?

Reply:
Yes. Domains are always in cache